### ISSUE TITLE
Invalid values added to autocomplete for tsx components with number or string literal typings

### ISSUE THREAD

**Issue by marbemac** [microsoft/TypeScript#53916]
Using properties in `tsx` components whos typing is a literal string, number, or union, results in all of those values being added to the top of autocomplete suggestions.

Code sandbox example: https://codesandbox.io/s/tsx-props-union-type-autocomplete-bug-z740bd?file=/src/App.tsx

Code:

```tsx
interface CompBProps {
  name?: string;
  propA?: "1" | "2" | "3" | "4" | "5" | "6" | "7" | "8" | "9";
}

const CompB = (props: CompBProps) => <div />;

export default function CompA() {
  return <CompB propA="1"  />;
}
```

Autocomplete works as expected if triggered before the prop:

<img width="635" alt="Screenshot 2023-04-19 at 11 42 23 AM" src="https://user-images.githubusercontent.com/847542/233144221-0f1811c1-7ed6-4582-82b0-196e8ed853f8.png">

If triggered after any props with literal types, all of those values are included in the autocomplete (and actually selecting any of these suggestions results in broken code).

<img width="929" alt="Screenshot 2023-04-19 at 11 42 42 AM" src="https://user-images.githubusercontent.com/847542/233144450-b76a595f-abb7-49c1-a99c-37db7baa146f.png">

Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: 1.78.0-insider and 1.77.3
- OS Version: OSX 13.1

Steps to Reproduce:

1. Create a `.tsx` file (I have tested in react and solidjs projects, doesn't seem to matter)
2. Define a component (CompA) that accepts a prop (propA) that is a string or number literal (or union type)
3. In a separate component (CompB), use (CompA). Note that before adding (propA), the autocomplete suggestions work as expected.
4. Set (propA), note now how triggering autocomplete after (propA) includes all of the possible values from (propA). Triggering autocomplete before (propA) still works as expected.

**Comment by Andarist** [comment]
This was already fixed by https://github.com/microsoft/TypeScript/pull/52676 so 5.1 doesn't suffer from it.

**Comment by marbemac** [comment]
Can confirm, fixed in 5.1 🎉.
